src/App.tsx

```diff
@@ -15,48 +15,102 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  reverse: boolean,
+  type: number,
+};
+
+function ReorderGoods(goods: string[], { reverse, type }: ReorderOptions) {
+  const initialGoods = [...goods];
+
+  switch (type) {
+    case 1:
+      initialGoods.sort((a, b) => a.localeCompare(b));
+      break;
+
+    case 2:
+      initialGoods.sort((a, b) => a.length - b.length);
+      break;
+
+    default:
+      break;
+  }
+
+  if (reverse) {
+    initialGoods.reverse();
+  }
+
+  return initialGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortType, useType] = useState(SortType.NONE);
+  const [reverse, useReverse] = useState(false);
```

Comment:
```suggestion
  const [sortType, setSortType] = useState(SortType.NONE);
  const [isReversed, setIsReversed] = useState(false);
```
Pay attention to the naming
https://react.dev/reference/react/useState